Source/Core/Core/BootManager.cpp

```diff
@@ -378,9 +378,9 @@ bool BootCore(const std::string& _rFilename)
   // Apply overrides
   // Some NTSC GameCube games such as Baten Kaitos react strangely to
   // language settings that would be invalid on an NTSC system
-  if (!StartUp.bOverrideGCLanguage && ntsc)
+  if (!StartUp.bWii && !StartUp.bOverrideGCLanguage && !StartUp.GCLanguageAndRegionCompatible())
```

Comment:
It's intended for the GameCube language setting only, but that setting does exist in Wii mode too (and Wii games don't use it). Note that `SelectedLanguage` only is for the GC language even though the variable name doesn't say so.